src/SmartComponents/SystemsTable/hooks.js

```diff
@@ -275,10 +295,15 @@ export const useSystemBulkSelect = ({
 
   const onSelectCallback = async (selectedIds) => {
     dispatch(setDisabledSelection(true));
-    const systemsSelection = loadedItems.filter(({ id }) =>
-      selectedIds.includes(id)
+    const systemsToFetch = selectedIds.filter((id) =>
+      loadedItems.map(({ id }) => id).includes(id)
     );
-    onSelect && onSelect(systemsSelection);
+
+    const fetchedSystems = (await fetchSystemsById(systemsToFetch)).map(
+      ({ data }) => data
+    );
+
+    onSelect && onSelect([...loadedItems, ...fetchedSystems]);
```

Comment:
This makes all items on the page to be selected if you check just one system. I believe we need to rewrite it to
```suggestion
    onSelect && onSelect(fetchedSystems);
```

Reply:
Should be fixed now. Only taking the fetchedSystems would then in other conditions cause the items on the current page not being properly added. We do need the `lodedItems`, but we need to filter them be the ids that have actually been selected.

Reply:
Honestly, I'm not even sure why we do it this complicated. As far as I can tell the batch request seems to have the same data that a separate request for each system returns. And the way this is implemented doesn't seem right. The return and the call to a state update are a 🚩.